626 canvas 2

tvay11 opened the pull request:
<img width="1440" alt="Screen Shot 2022-10-11 at 9 28 28 PM" src="https://user-images.githubusercontent.com/83725863/195235199-ba239f3f-ade9-46bc-873f-e19cfc36baa6.png">

This allows it to open "TPEN/canvas/" files. This updates buildpage in CanvasServlet.java to build page in Canvas.

thehabes commented on src/java/edu/slu/tpen/servlet/CanvasServlet.java:
```diff
@@ -100,76 +103,88 @@ public String getServletInfo() {
-                    canvasWidth = storedDims.width * canvasHeight / storedDims.height; 
-                    createFolioDimsRecord(storedDims.width, storedDims.height, canvasWidth, canvasHeight, f.getFolioNumber());
-                }
+     private JSONObject buildPage(int projID, Folio f) throws SQLException, IOException {
```
This is not the right build page.  You are not getting the Annotation lines.  You need to use the same buildPage that JSONLDExporter.java is using.

thehabes commented on src/java/edu/slu/tpen/servlet/CanvasServlet.java:
```diff
@@ -100,76 +103,88 @@ public String getServletInfo() {
+            //System.out.println("Get otherContent");
+            //System.out.println(projID + "  " + canvasID + "  " + f.getFolioNumber() + "  " + u.getUID());;
+            //otherContent = getAnnotationListsForProject(-, canvasID, 0);
+            otherContent = getLinesForProject(projID, canvasID, f.getFolioNumber(), 0); //Can be an empty array now.
```
This does not actually contain the lines.  See the example below.  `resources` is an empty array.

```JSON
{
   "@wid":"http://localhost:8080/TPEN/canvas/13251387",
   "@type":"sc:Canvas",
   "label":"IP_001_001",
   "width":749,
   "height":1000,
   "otherContent":[
      {
         "@id":"http://localhost:8080/TPEN/project/13251387/annotations/13251387",
         "@type":"sc:AnnotationList",
         "label":"http://localhost:8080/TPEN/canvas/13251387 List",
         "proj":13251387,
         "on":"http://localhost:8080/TPEN/canvas/13251387",
         "@context":"http://iiif.io/api/presentation/2/context.json",
         "resources":[
            
         ]
      }
   ],
   "images":[
      {
         "@type":"oa:Annotation",
         "motivation":"sc:painting",
         "resource":{
            "@id":"https://iiif.library.utoronto.ca/image/v2/italianpaleography:IP_001_001/full/full/0/default.jpg",
            "@type":"dctypes:Image",
            "format":"image/jpeg"
         },
         "on":"http://localhost:8080/TPEN/canvas/13251387"
      }
   ]
}
```

Here is that same Canvas returned by the /manifest/ servlet.  Note this is the manifest your servlet should be returning.  Test your code by comparing to the canvases in /TPEN/manifest/7211
```JSON
{
   "@id":"http://localhost:8080/TPEN/canvas/13251387",
   "@type":"sc:Canvas",
   "label":"IP_001_001",
   "width":749,
   "height":1000,
   "otherContent":[
      {
         "@id":"http://localhost:8080/TPEN/project/7211/annotations/13251387",
         "@type":"sc:AnnotationList",
         "label":"http://localhost:8080/TPEN/canvas/13251387 List",
         "proj":7211,
         "on":"http://localhost:8080/TPEN/canvas/13251387",
         "@context":"http://iiif.io/api/presentation/2/context.json",
         "resources":[
            {
               "@id":"http://localhost:8080/TPEN/line/104720616",
               "_tpen_line_id":"line/104720616",
               "@type":"oa:Annotation",
               "motivation":"oad:transcribing",
               "resource":{
                  "@type":"cnt:ContentAsText",
                  "cnt:chars":"This is the first line"
               },
               "on":"http://localhost:8080/TPEN/canvas/13251387#xywh=41,64,629,66",
               "_tpen_note":"",
               "_tpen_creator":11,
               "modified":"2022-09-12 13:21:09.0"
            },
            {
               "@id":"http://localhost:8080/TPEN/line/104720617",
               "_tpen_line_id":"line/104720617",
               "@type":"oa:Annotation",
               "motivation":"oad:transcribing",
               "resource":{
                  "@type":"cnt:ContentAsText",
                  "cnt:chars":"This is the second line"
               },
               "on":"http://localhost:8080/TPEN/canvas/13251387#xywh=41,130,629,27",
               "_tpen_note":"",
               "_tpen_creator":11,
               "modified":"2022-09-12 13:21:12.0"
            },
            {
               "@id":"http://localhost:8080/TPEN/line/104720618",
               "_tpen_line_id":"line/104720618",
               "@type":"oa:Annotation",
               "motivation":"oad:transcribing",
               "resource":{
                  "@type":"cnt:ContentAsText",
                  "cnt:chars":"This is the third line"
               },
               "on":"http://localhost:8080/TPEN/canvas/13251387#xywh=41,157,629,19",
               "_tpen_note":"",
               "_tpen_creator":11,
               "modified":"2022-09-12 13:21:15.0"
            },
            {
               "@id":"http://localhost:8080/TPEN/line/104720619",
               "_tpen_line_id":"line/104720619",
               "@type":"oa:Annotation",
               "motivation":"oad:transcribing",
               "resource":{
                  "@type":"cnt:ContentAsText",
                  "cnt:chars":"This is the fourth line"
               },
               "on":"http://localhost:8080/TPEN/canvas/13251387#xywh=41,176,629,16",
               "_tpen_note":"",
               "_tpen_creator":11,
               "modified":"2022-09-12 13:21:18.0"
            },
            {
               "@id":"http://localhost:8080/TPEN/line/104720620",
               "_tpen_line_id":"line/104720620",
               "@type":"oa:Annotation",
               "motivation":"oad:transcribing",
               "resource":{
                  "@type":"cnt:ContentAsText",
                  "cnt:chars":"This is the fifth line"
               },
               "on":"http://localhost:8080/TPEN/canvas/13251387#xywh=41,193,629,31",
               "_tpen_note":"",
               "_tpen_creator":11,
               "modified":"2022-09-12 13:21:21.0"
            },
            {
               "@id":"http://localhost:8080/TPEN/line/104720621",
               "_tpen_line_id":"line/104720621",
               "@type":"oa:Annotation",
               "motivation":"oad:transcribing",
               "resource":{
                  "@type":"cnt:ContentAsText",
                  "cnt:chars":""
               },
               "on":"http://localhost:8080/TPEN/canvas/13251387#xywh=41,224,629,232",
               "_tpen_note":"",
               "_tpen_creator":11,
               "modified":"2022-09-12 11:35:20.0"
            }
         ]
      }
   ],
   "images":[
      {
         "@type":"oa:Annotation",
         "motivation":"sc:painting",
         "resource":{
            "@id":"https://iiif.library.utoronto.ca/image/v2/italianpaleography:IP_001_001/full/full/0/default.jpg",
            "@type":"dctypes:Image",
            "format":"image/jpeg"
         },
         "on":"http://localhost:8080/TPEN/canvas/13251387"
      }
   ]
}
```

thehabes commented on src/java/edu/slu/tpen/servlet/CanvasServlet.java:
```diff
@@ -100,76 +103,88 @@ public String getServletInfo() {
+            }
+            LOG.log(INFO, "pageDim={0}", pageDim);
+            JSONObject result  = new JSONObject();
+            result.put("@wid", canvasID);
```
No.  You know what you did.

thehabes reviewed (changes requested):
Please remove the .idea directory from the pull request and add it to your .gitignore.  
Please remove nbbuild.xml from the pull request and add it to your .gitignore

If you commit these, other people will have to pull them down and it will mess with their environment.  Keep environment specific things local.

The value for `proj` in your canvases in incorrect.  See below

**Your Canvas**
```JSON
{
   "@wid":"http://localhost:8080/TPEN/canvas/13251387",
   "@type":"sc:Canvas",
   "label":"IP_001_001",
   "width":749,
   "height":1000,
   "otherContent":[
      {
         "@id":"http://localhost:8080/TPEN/project/13251387/annotations/13251387",
         "@type":"sc:AnnotationList",
         "label":"http://localhost:8080/TPEN/canvas/13251387 List",
         "proj":13251387,
         .
         .
         .
```
**/manifest/7211 Canvas**
```JSON
{
   "@wid":"http://localhost:8080/TPEN/canvas/13251387",
   "@type":"sc:Canvas",
   "label":"IP_001_001",
   "width":749,
   "height":1000,
   "otherContent":[
      {
         "@id":"http://localhost:8080/TPEN/project/13251387/annotations/13251387",
         "@type":"sc:AnnotationList",
         "label":"http://localhost:8080/TPEN/canvas/13251387 List",
         "proj":7211,
         .
         .
         .
```

thehabes commented:
Hey @tvay11,

After looking at this a bit more, some of these issues are things that have actually changed.  Typically, the Canvas knows the projectID it belongs to, and the lines know they belong to the Canvas for that project.  However, with the /canvas/ servlet, we have removed the layer that knew about the project.  This has some implications for what you can return.

- There are many AnnotationLists containing lines for this Canvas, and each one belongs to their own project.  This Canvas will not actually know which lines to pick, unless the URI contains the project id somehow like `http://localhost:8080/TPEN/project/7211/canvas/13251387` which isn't really what we want.  Therefore, I believe this particular canvas serlvet should not return any lines until we decide what we want to do about it.
- For the same reason, I do not think you need to have the `proj` property either.

I think an output like the following should be what you aim for with Presentation API 2.

```JSON
{
   "@context":"http://iiif.io/api/presentation/2/context.json",
   "@id":"http://localhost:8080/TPEN/canvas/13251387",
   "@type":"sc:Canvas",
   "label":"IP_001_001",
   "width":749,
   "height":1000,
   "images":[
      {
         "@type":"oa:Annotation",
         "motivation":"sc:painting",
         "resource":{
            "@id":"https://iiif.library.utoronto.ca/image/v2/italianpaleography:IP_001_001/full/full/0/default.jpg",
            "@type":"dctypes:Image",
            "format":"image/jpeg"
         },
         "on":"http://localhost:8080/TPEN/canvas/13251387"
      }
   ]
}
```

tvay11 commented:
Fixed gitignore and Changed buildpage to  look presentation API 2.

![](file:///var/folders/k0/y4rsv_nx1rxby2kdwl0bnht40000gn/T/TemporaryItems/NSIRD_screencaptureui_mY88WV/Screen%20Shot%202022-10-17%20at%208.04.56%20PM.png)

tvay11 commented:
<img width="1440" alt="Screen Shot 2022-10-17 at 8 04 24 PM" src="https://user-images.githubusercontent.com/83725863/196312068-5c9befb0-d5d4-42de-b041-1278bba34fba.png">

tvay11 commented:
<img width="716" alt="Screen Shot 2022-10-17 at 8 12 58 PM" src="https://user-images.githubusercontent.com/83725863/196312172-2d443af5-6a28-4021-a506-f2f939c65ba9.png">

cubap commented on TPEN28.iml:
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
We don't need this iml file. It is specific to your IDE.

cubap commented on nbproject/ant-deploy.xml:
```diff
@@ -0,0 +1,57 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
We do not need anything in the nbproject directory. You deleted that line above.

cubap commented on src/java/version.properties:
```diff
@@ -0,0 +1,30 @@
+# Create internal locations in version.properties
```
nope nope nope. Get this outta here.

thehabes commented on .gitignore:
```diff
@@ -38,15 +21,32 @@ $RECYCLE.BIN/
 Temporary Items
 .apdisk
-nbproject
-src/java/version.properties
```
Yes, you will want to keep version.properties in this gitignore.  I tried to give a live time demo, but had trouble because the app wouldn't start.  I pulled down your code, and my properties file was overwritten, which broke my environment.  Big no-no.

tvay11 replied:
I apologize for not including it in my gitignore.

thehabes commented:
You need to fix your .gitignore, as mentioned by Patrick above.

One more thing for you to think about.  http://localhost:8080/TPEN/canvas/0010010101 passes as "successful" with the code 200 (SC_OK) and returns and empty object.

In the case where there is no folio for a given folio ID, the servlet should fail with code 404 (SC_NOT_FOUND).  It is not necessary to do as a part of this pull request.  I will make an issue for this.

thehabes commented on src/java/edu/slu/tpen/servlet/CanvasServlet.java:
```diff
@@ -7,16 +7,20 @@
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
-import static edu.slu.tpen.entity.Image.Canvas.getAnnotationListsForProject;
+import static edu.slu.tpen.entity.Image.Canvas.getLinesForProject;
```
Nice, good catch.  We aren't going to do lines anymore though, so now this is unused.  Just remove it.

thehabes commented on src/java/edu/slu/tpen/servlet/CanvasServlet.java:
```diff
@@ -49,7 +52,7 @@ public class CanvasServlet extends HttpServlet{
      */
```
Arrghh the old documentation was wrong.  Can you update this bit?
"serialisation of the requested T-PEN **canvas**."

thehabes reviewed (changes requested):
The `nbproject` folder is still present.  Try to get it out of the PR.  Even if it is in your .gitignore, you might still need to ["unstage" it](https://devconnected.com/how-to-unstage-files-on-git/#:~:text=To%20unstage%20commits%20on%20Git,and%20specify%20the%20commit%20hash.&text=Alternatively%2C%20if%20you%20want%20to%20unstage%20your%20last%20commit%2C%20you,order%20to%20revert%20it%20easily.&text=Using%20the%20%E2%80%9C%E2%80%93soft%E2%80%9D%20argument,your%20working%20directory%20and%20index.).

thehabes reviewed (approved):
Looks good to go!

Woodlnsoul reviewed (commented):
Looks good!
